p2p/host/autorelay/relay_finder.go

```diff
@@ -699,11 +718,17 @@ func (rf *relayFinder) selectCandidates() []*candidate {
 		}
 	}
 
-	// TODO: better relay selection strategy; this just selects random relays,
-	// but we should probably use ping latency as the selection metric
-	rand.Shuffle(len(candidates), func(i, j int) {
-		candidates[i], candidates[j] = candidates[j], candidates[i]
-	})
+	// If the node speed test is enabled, the rtt will be sorted,
+	// otherwise the order will be randomly shuffled.
+	if rf.sortByRTT {
+		sort.Slice(candidates, func(i, j int) bool {
```

Comment:
this should be fuzzy, otherwise it can create hot spots in the network.

Reply:
> this should be fuzzy, otherwise it can create hot spots in the network.

Sorry, I didn't quite get your point. Could you be a little more detailed?

Reply:
fuzz the RTTs, if they are exact then well positioned nodrs in the network might become hot spots.

Reply:
it is true that those high quality nodes in the network are preferred, but the node has already shown that he can improve the relay service for us when he accepted to be able to be our candidate.

Reply:
I agree with Vyzo. We don't want to introduce a change that could cause all well-behaved nodes to possibly DoS a well behaved relay.

Reply:
We do not enable rtt speed measurement by default, which should alleviate this problem.

Regarding the DoS issue, I think we should protect the server side instead of the client side. The libp2p code itself is open source, and anyone can change the behavior of the libp2p relay client, so restricting it here is useless for hackers who want to perform DoS attacks on our public network nodes.

Reply:
There's a difference between a malicious user doing something bad with nodes they control and us shipping code that causes *every* node to behave badly.